hooks/useSendMessage.ts

```diff
@@ -0,0 +1,33 @@
+import { Conversation } from '@xmtp/xmtp-js'
+import { useAppStore } from '../store/app'
+
+const useSendMessage = (selectedConversation?: Conversation) => {
+  const client = useAppStore((state) => state.client)
+  const peerAddress = selectedConversation?.peerAddress
+  const conversationId = selectedConversation?.context?.conversationId
+
+  const sendMessage = async (message: string) => {
+    if (!client || !peerAddress) {
+      return
+    }
+    let conversation
+    if (conversationId) {
+      conversation = await client.conversations.newConversation(peerAddress, {
+        conversationId: conversationId,
+        metadata: {},
+      })
+    } else {
+      conversation = await client.conversations.newConversation(peerAddress)
+    }

```

Comment:
do we have to create a new conversation here if the selected conversation already exists? it seems like in the first use case we're passing the same convo id in so it seems like it could be reused perhaps?

Reply:
Great call out. With v7 of the SDK creating a new conversation is slow and expensive. We should figure out ways of passing them around.